### PR TITLE
DSDEEPB-489 and DSDEEPB-1618 for UI functionality and for a checkbox for public readability (depending on checked status) 

### PR DESCRIPTION
**Pull request by eddiebroad**
have UI report what is retireved from ORCH
set up for add new users to ACL

added helper utilities for retrieving public status

add random public user to endpoints

add filter utility and make the UI still represent values

put in a check to make sure user does not enter public for user

cleanup and prep for saving to back end

### CODE REVIEW

**Comment by rushtong** [comment]
Tested functionality and it seems to work as advertised.

**Comment by rushtong** [comment]
There are a lot of cases where a single plain string is repeated, i.e. "READER", "OWNER", etc. Can these be refactored into a reusable object in some way? It would reduce copy/paste error and allow for easier updates when new or changed ACLs come along.

**Comment by rushtong** [comment]
Done with review. Looks good as far as my clojure knowledge goes.

File: src/cljs/org/broadinstitute/firecloud_ui/page/methods_configs_acl.cljs [hunk 1]
```diff
@@ -21,11 +21,66 @@
 (def ^:private access-levels
   ["READER" "OWNER" "NO ACCESS"])
 
+(def ^:private reader-level (nth access-levels 0))
+(def ^:private  owner-level (nth access-levels 1))
+(def ^:private  no-access-level (nth access-levels 2))
```
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/page/methods_configs_acl.cljs]: Extra spaces

File: src/cljs/org/broadinstitute/firecloud_ui/page/methods_configs_acl.cljs [hunk 2]
```diff
@@ -21,11 +21,66 @@
...
+
+(defn- index-to-access-level [idx]
+  (str (nth access-levels idx)))
```
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/page/methods_configs_acl.cljs]: Unnecessary "str"

File: src/cljs/org/broadinstitute/firecloud_ui/page/methods_configs_acl.cljs [hunk 3]
```diff
@@ -21,11 +21,66 @@
...
+
 (def ^:private column-width "calc(50% - 4px)")
 
+(defn- correspondsToReader [access-level]
+  (if
+    (or
+      (= access-level reader-level)
+      (= access-level owner-level))
+    true
+    false))
```
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/page/methods_configs_acl.cljs]: `(if <cond> true false)` => `<cond>`

File: src/cljs/org/broadinstitute/firecloud_ui/page/methods_configs_acl.cljs [hunk 4]
```diff
@@ -21,11 +21,66 @@
...
+
+(defn- filter-public [acl-vec]
+  (let [hasNotPublicUser
+        (fn [m]
+          (not
+            (and
+              (map? m)
+              (contains? m :user)
```
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/page/methods_configs_acl.cljs]: Do we really need to be testing whether the item is a map and conforms to the expected schema?  Seems like upstream code should guarantee this.

File: src/cljs/org/broadinstitute/firecloud_ui/page/methods_configs_acl.cljs [hunk 5]
```diff
@@ -21,11 +21,66 @@
...
+(defn- make-ui-vec [a-map]
+  {:user (get a-map "user")
+   :role (get a-map "role")})
+
+(defn- extract-last-public-access-level [acl-vec]
+  (let [hasPublicUser
+        (fn [m]
+          (and
+            (map? m)
+            (contains? m :user)
```
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/page/methods_configs_acl.cljs]: Same comment as above.  Also, these functions are just inverses of each other and could be factored out.

File: src/cljs/org/broadinstitute/firecloud_ui/page/methods_configs_acl.cljs [hunk 6]
```diff
@@ -60,19 +115,33 @@
...
                               :spellCheck false
-                              :defaultValue (acl-entry "user")})
+                              :defaultValue (:user acl-entry)
+                              :onChange (fn [e]
+                                          (let [new-val (-> e .-target .-value)
+                                                new-val-is-public (= "public" new-val)]
+                                            (when new-val-is-public
+                                              (do
+                                                (js/alert "Cannot set value to 'public'!  Use the check-box instead.")
+                                                (set! (-> e .-target .-value) "")))))})
```
Comment (MattPutnam) [inline src/cljs/org/broadinstitute/firecloud_ui/page/methods_configs_acl.cljs]: As with the previous PR, don't do this.  Validate the text fields when the OK button is pressed.  What if someone's ID starts with "public"?

**Pull request merged**